extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -473,30 +488,36 @@ export class AmpStoryReaction extends AMP.BaseElement {
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.backendDataPromise_.then(() => {
-      if (this.hasUserSelection_) {
-        return;
-      }
-
-      this.triggerAnalytics_(optionEl);
-      this.hasUserSelection_ = true;
+    this.backendDataPromise_
+      .then(() => {
+        if (this.hasUserSelection_) {
+          return;
+        }
 
-      if (this.optionsData_) {
-        this.optionsData_[optionEl.optionIndex_]['totalCount']++;
-        this.optionsData_[optionEl.optionIndex_]['selectedByUser'] = true;
-      }
+        this.triggerAnalytics_(optionEl);
+        this.hasUserSelection_ = true;
 
-      this.mutateElement(() => {
         if (this.optionsData_) {
-          this.updateOptionPercentages_(this.optionsData_);
+          this.optionsData_[optionEl.optionIndex_]['totalCount']++;
+          this.optionsData_[optionEl.optionIndex_]['selectedByUser'] = true;
         }
-        this.updateToPostSelectionState_(optionEl);
-      });
 
-      if (this.element.hasAttribute('endpoint')) {
-        this.executeReactionRequest_('POST', optionEl.optionIndex_);
-      }
-    });
+        this.mutateElement(() => {
+          if (this.optionsData_) {
+            this.updateOptionPercentages_(this.optionsData_);
+          }
+          this.updateToPostSelectionState_(optionEl);
+        });
+
+        if (this.element.hasAttribute('endpoint')) {
+          this.executeReactionRequest_('POST', optionEl.optionIndex_);
+        }
+      })
+      .catch(() => {
+        this.mutateElement(() => {
+          this.updateToPostSelectionState_(optionEl);
```

Comment:
Can you give some context on this?

Reply:
Since this handles the option selection, when the backend was not properly connected (eg. I didn't have my backend running), the components with endpoint would not respond to user clicking and never update to post selection state. This ensures that the backendDataPromise failing still shows the results as if it was a local reaction (no percentages but still updating the states). So now, the component is more stable to backend/connection errors.